### PR TITLE
Skeleton Package Work

### PR DESCRIPTION
**Pull request by ConnorVG**
This performs some skeleton package work, including:
 - `.editorconfig` – optional, happy to remove, helps keep common development strategies;
 - `.gitattributes` + `.gitignore` – Development and composer installation improvements;
 - `.travis.yml` – Travis support;
 - `phpunit.xml` + `tests/` – PHPUnit integration;
 - `composer.json` – tweaks to structure, improve definition and autoload tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/vendor
+composer.phar
+composer.lock
```
Comment (ntzm) [inline .gitignore]: the lock file shouldn't be ignored for projects right?

Comment (MatthewSH) [inline .gitignore]: I don't think so.

Comment (ConnorVG) [inline .gitignore]: This isn't a project, it's a package.

Comment (ntzm) [inline .gitignore]: I'm stupid

Comment (MatthewSH) [inline .gitignore]: Hm, learning a lot today! Thanks @ConnorVG

File: composer.json [hunk 2]
```diff
@@ -12,11 +12,30 @@
     ],
     "require": {
         "php": ">=7.0.0",
-        "illuminate/database": "5.4.*"
+        "illuminate/database": "~5.4.0"
+    },
+    "require-dev": {
+        "mockery/mockery": "^0.9",
+        "phpunit/phpunit": "^4.0|^5.0"
```
Comment (svenluijten) [inline composer.json]: Why don't you pull in phpunit version 6?

**Comment by ConnorVG** [comment]
It’s a bit of a bitch to work with, especially when you already have
Laravel in-scope (it uses PHPUnit 5.7). This is all preference though. :)

Cheers,
 – Connor S. Parks

On 1 June 2017 at 10:17:19, Sven Luijten (notifications@github.com) wrote:

*@svenluijten* commented on this pull request.
------------------------------

In composer.json
<https://github.com/AndreasElia/laravel-forum/pull/4#discussion_r119565426>:

> @@ -12,11 +12,30 @@
     ],
     "require": {
         "php": ">=7.0.0",
-        "illuminate/database": "5.4.*"
+        "illuminate/database": "~5.4.0"
+    },
+    "require-dev": {
+        "mockery/mockery": "^0.9",
+        "phpunit/phpunit": "^4.0|^5.0"

Why don't you pull in phpunit version 6?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/AndreasElia/laravel-forum/pull/4#pullrequestreview-41455470>,
or mute the thread

**Comment by Dylan-DPC** [comment]
Shift to PHPUnit 6. I have used it on another project there are no issues with it

File: tests/BaseTest.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace AndreasElia\Tests\Forum;
+
+use Mockery;
+use PHPUnit_Framework_TestCase;
+
+abstract class BaseTest extends PHPUnit_Framework_TestCase
```
Comment (Dylan-DPC) [inline tests/BaseTest.php]: change to PHPUnit 6 testcase class

**Pull request merged**